headers/analog_driver.h

```diff
@@ -0,0 +1,538 @@
+ *          within the sequence and total sequence length/size. 
+ */
+typedef enum {
+    ADC_SEQ_1 = 1, 
+    ADC_SEQ_2,  
+    ADC_SEQ_3,  
+    ADC_SEQ_4,  
+    ADC_SEQ_5,  
+    ADC_SEQ_6,  
+    ADC_SEQ_7,  
+    ADC_SEQ_8,  
+    ADC_SEQ_9,  
+    ADC_SEQ_10, 
+    ADC_SEQ_11, 
+    ADC_SEQ_12, 
+    ADC_SEQ_13, 
+    ADC_SEQ_14, 
+    ADC_SEQ_15, 
+    ADC_SEQ_16 
+} adc_seq_num_t; 
+
+
+/**
+ * @brief EOC selection 
+ */
+typedef enum {
+    ADC_EOC_SEQ,   // EOC bit set after each sequence of regular conversions 
+    ADC_EOC_EACH   // EOC bit set after each regular conversion 
+} adc_eoc_config_t; 
+
+
+/**
+ * @brief EOC interrupt selection 
+ */
+typedef enum {
+    ADC_EOC_INT_DISABLE,   // Disable EOC interrupt 
+    ADC_EOC_INT_ENABLE     // Enable EOC interrupt 
+} adc_eoc_int_t; 
+
+
+/**
+ * @brief Scan mode selection 
+ */
+typedef enum {
+    ADC_SCAN_DISABLE,   // Disable scan mode 
+    ADC_SCAN_ENABLE     // Enable scan mode 
+} adc_scan_t; 
+
+
+/**
+ * @brief Continuous mode selection 
+ */
+typedef enum {
+    ADC_CONT_DISABLE,   // Disable continuous mode (single conversion mode) 
+    ADC_CONT_ENABLE     // Enable continuous mode 
+} adc_cont_t; 
+
+
+/**
+ * @brief DMA mode selection 
+ */
+typedef enum {
+    ADC_DMA_DISABLE,   // Disable DMA for ADC 
+    ADC_DMA_ENABLE     // Enable DMA for ADC 
+} adc_dma_t; 
+
+
+/**
+ * @brief DMA Disable selection 
+ */
+typedef enum {
+    ADC_DDS_DISABLE,    // No new DMA request is issued after the last transfer 
+    ADC_DDS_ENABLE      // DMA requets are issued as long as data are coverted 
+} adc_dds_t; 
+
+
+/**
+ * @brief Watchdog selection 
+ */
+typedef enum {
+    ADC_WD_DISABLE,   // Disable the watchdog 
+    ADC_WD_ENABLE     // Enable the watchdog 
+} adc_wd_t; 
+
+
+/**
+ * @brief Watchdog single channel selection 
+ */
+typedef enum {
+    ADC_WD_SC_DISABLE,   // Disable the watchdog single channel selection 
+    ADC_WD_SC_ENABLE     // Enable the watchdog single channel selection 
+} adc_wd_sc_t; 
+
+
+/**
+ * @brief Watchdog interrupt selection 
+ */
+typedef enum {
+    ADC_WD_INT_DISABLE,   // Disable the watchdog interrupt 
+    ADC_WD_INT_ENABLE     // Enable the watchdog interrupt 
+} adc_awdie_t; 
+
+
+/**
+ * @brief Overrun interrupt selection 
+ */
+typedef enum {
+    ADC_OVR_INT_DISABLE,   // Disable the overrun interrupt 
+    ADC_OVR_INT_ENABLE     // Enable the overrun interrupt 
+} adc_ovrie_t; 
+
+//================================================================================
+
+
+//================================================================================
+// Initialization 
+
+/**
+ * @brief ADC port initialization 
+ * 
+ * @details This functions defines the behavior of an ADC port (ex. ADC1). All the channels 
+ *          within the ADC port follow this configuration. This function gets called once 
+ *          for each port. See the enum listed below for argument inputs. 
+ * 
+ * @see adc_prescalar_t
```

Comment:
Super nit:

Never seen this @see before. 

I would say it is not common practice to list out the types used.